### ISSUE TITLE
Album LIst - option to apply filter to songs in the playlist

### ISSUE THREAD

**Issue by davidSchlosser** [quodlibet/quodlibet#2484]
With the Album List  browser, the filter is applied only to album list, similarly for Cover Grid & Album Collection browsers. I tend to use the Album List browser nearly all the time as I always want to select from the albums that include tracks that match my filter.

However, for the ways I use filters, this could be way more useful if the filter also applied to the song list in the common case of albums which contain tracks with various genre, artist, performer and other tags. (I like to filter on a custom 'mood' tag on all my tracks and most albums contain tracks with a mix of different moods).

Realising that many users must be happy with the way filters currently work with these browsers, I'd hugely welcome an enhancement that allows me to extend the filter to apply also the song list.

Alternatively, a separate filter for the song list would provide the capability.

Thanks

**Comment by declension** [comment]
Hmm. I feel this is in danger of confusing the already complex mental model, though I also see the use case you have here.

Remember also the filtering done by albums (or more generally _collections_) is not entirely compatible with the song filtering e.g. `#(tracks>4)` or `#(length:min>60)`. We'd probably just want to (somehow) ignore meaningless queries at a song level if doing this.

But yes I feel the UI is key here. A single checkbox / radio button of "filter by album" vs "filter by songs" might be nice enough but it should probably be applied to all the collection browsers that can select multiple collections (so _not_ Playlists currently, though that browser actually does this by default, come to think of it. Hmmm).

**Comment by davidSchlosser** [comment]
The current behaviour of filters is already somewhat confusing as it depends on the browser. So I'd welcome it being made more predictable. It's unexpected to me currently that, unlike the filter behaviour in the Album List browser, a custom Paned Browser with just an Album pane, filters songs, not albums.

The use case I gave would work ok with the single checkbox/radio button, but I would prefer it set as a browser preference. That way I can have my Album List play filtering on songs, while I open a separate Album List browser and manipulate all songs for my filtered album (I have a frequent need to view &/or modify tags of all songs of albums I've recently played).

**Comment by davidSchlosser** [comment]
Sorry - I didn't mean to close this issue.

**Comment by declension** [comment]
Yes actually I meant conceptually the distinction should be added for all those browser rather than a single global preference, ie agree that per instance is probably best.

I was envisaging a drop down near the query combobox(es).

Getting this right would probably need a bit of experimentation either way..

**Comment by davidSchlosser** [comment]
Not sure what you mean by 'query comboboxes'.

Yes, tricky. With 'filter by albums' vs 'filter by songs' I would infer that with 'filter by songs' ALL albums would show in the album pane, including those with no songs matching the filter. That wouldn't be great for my cases.

What about a checkbox in the Options part of the relevant browser Preferences dialogue? The Album List Preferences already includes a checkbox for search ('Inline search includes people'). The Album List, Cover Grid & Album Collection Preferences options might include a checkbox 'Extend search filter to songs'.

(Incidentally, #(tracks>4) seems to filter well on the 'Search library' browser.)

**Comment by declension** [comment]
OK. So something like _only show songs that match_ might be clearer?

Remember though that browser preferences are across _all_ instances of that browser (well in theory at least, depends how they're managing the config), hence my thinking this might be best as part of the visible UI when you type the query...

**Comment by davidSchlosser** [comment]
Yes, that's clear.
I agree with your thinking about not using browser preferences.
